src/plugins/credential_management/public/components/utils.ts

```diff
@@ -44,8 +44,9 @@ export async function deleteCredentials(
   savedObjectsClient: SavedObjectsClientContract,
   selectedCredentials: CredentialsTableItem[]
 ) {
-  // TODO: Refactor with nonblocking IO
-  for (const credential of selectedCredentials) {
-    await savedObjectsClient.delete('credential', credential.id);
-  }
+  await Promise.all(
```

Comment:
please add exceptional handling and friendly error message

Reply:
Added in the listing page

Reply:
@mpabba3003  @kristenTian do you guy know what is the best practice of handling errors on UI? For example, in such utility function which may get API errors, does it usually do nothing about error handling and relies on the corresponding UI component to catch and handle the error like this, or util functions should throw some concrete error explicitly, or is there other approach to better handling this?

my concern here is that, this util function may throw an error, but UI components that uses this function may have no idea by looking at this function code, as it doesn't throw any error, thus other UI component author may miss error handling. Any suggestsion?

Reply:
> @mpabba3003 @kristenTian do you guy know what is the best practice of handling errors on UI? For example, in such utility function which may get API errors, does it usually do nothing about error handling and relies on the corresponding UI component to catch and handle the error like this, or util functions should throw some concrete error explicitly, or is there other approach to better handling this?
> 
> my concern here is that, this util function may throw an error, but UI components that uses this function may have no idea by looking at this function code, as it doesn't throw any error, thus other UI component author may miss error handling. Any suggestsion?

The current error handling is consistent with both saved object and index pattern management. Another concern is that we should avoid displaying the stack trace on pages. I would recommend adding logs afterward, as @zhongnansu is doing research on.

Reply:
> 

i'm not sure current implementation aligned with IPM

this is what looks how IPM handle deletion. and there is confirmation dialog before actually deletion, i didn't  see that in current implementation. 
Please play IPM more and read through code

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d7004dc5b0392477fdd54ac66b29d231975a173b/src/plugins/index_pattern_management/public/components/edit_index_pattern/edit_index_pattern.tsx#L167

```
      overlays.openConfirm('', confirmModalOptionsDelete).then((isConfirmed) => {
        if (isConfirmed) {
          doRemove();
        }
      });
```

Reply:
It is better to have exception handlers .. Either try .. catch or error cases when promise is fulfilled .. 

In this case is our API not able to take multiple ids at a time and delete the saved credentials instead of making multiple calls to delete one at a time?

I suggest to have an API call that handles multi delete rather than making multiple calls

Reply:
> > 
> 
> i'm not sure current implementation aligned with IPM
> 
> this is what looks how IPM handle deletion. and there is confirmation dialog before actually deletion, i didn't see that in current implementation. Please play IPM more and read through code
> 
> https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d7004dc5b0392477fdd54ac66b29d231975a173b/src/plugins/index_pattern_management/public/components/edit_index_pattern/edit_index_pattern.tsx#L167
> 
> ```
>       overlays.openConfirm('', confirmModalOptionsDelete).then((isConfirmed) => {
>         if (isConfirmed) {
>           doRemove();
>         }
>       });
> ```

This is not the scope of this PR. It's tracked on #2055

Reply:
> It is better to have exception handlers .. Either try .. catch or error cases when promise is fulfilled ..

Yes it's at src/plugins/credential_management/public/components/credential_table/credentials_table.tsx line 141-164

Reply:
> In this case is our API not able to take multiple ids at a time and delete the saved credentials instead of making multiple calls to delete one at a time?

This is what the promise.all is doing - It's done on purpose because OpenSearch has no underpinning API support.